### ISSUE TITLE
Better Moments

### ISSUE THREAD

**Issue by melyux** [photoprism/photoprism#3055]
**Describe what problem this solves and why this would be valuable to many users**

Moments don't seem very useful at the moment. They group unrelated photos as "Parks", but that's a location categorization, not a moment. Those photos weren't taken during the same trip or near one another or even in the same year. That's just a label based on a very loose location keyword. Or Moments are created like "USA 2021" but an entire year is not a moment and the USA is a huge area. That's just a date.

**Describe the solution you'd like**

Something closer to how an iPhone decides on Moments so that they are useful for grouping together photos closely-related in time AND location with smaller granularity on both time and location. Like "New Year's in Canada" or "Trip to Chicago" or "Museum, December 12, 2022". This is probably not an exact science and AI is likely involved to make iPhone moments so good, but these are a couple hard and fast rules that all useful Moments should likely follow without requiring AI.

**Describe alternatives you've considered**

Hmm?

**Additional context**

Examples of current Moments created by PhotoPrism:

* Bays, Capes, & Beaches
* Nature & Landscape
* Outdoor Adventures
* Parks & Gardens (noticing a pattern here?)
* Water, Ships & Boats
* USA 2019
* USA 2020
* USA 2021
* USA 2022
* USA 2023

**Comment by graciousgrey** [comment]
> Moments don't seem very useful at the moment. They group unrelated photos as "Parks", but that's a location categorization, not a moment. Those photos weren't taken during the same trip or near one another or even in the same year.

Moments are not only meant for timely related trips but also for nice collections, such as "Bays, Capes, & Beaches". Apple also does not only create moments for trips but for other topics as well e.g. "Pets".

At the moment we do have different types of moments. Some are based on labels e.g. "Bays, Capes, & Beaches". Others are based on location and time. I totally agree that those can be refined :+1: 

Also adding more moments e.g. based on people such as "Family", "Jane & John" is something we would love to do.

**Comment by satheshshiva** [comment]
Some more suggestions based on Google Photos,

"They grow up faster" - Show picture of your kids from the past few years until recent one in an incremental basis.
"This day X years ago"

**Comment by lastzero** [comment]
AFAIK there's a separate issue for that already! Generally, improvements for the moments would be great thing to contribute ✨

**Comment by melyux** [comment]
I guess I'd like the "Moments" (e.g. a specific time and place) instead of the "Memories" (more generic, like "Bays" and "Outdoor"). The Moments in PhotoPrism are more akin to Apple's Memories instead of the Apple's Moments, which would be much more useful since they group all photos into clusters. That's something PhotoPrism needs imo

**Comment by lastzero** [comment]
Our moments include both, provided there any time & place clusters. Unless with time, you mean less than a month. It's unclear from what you wrote.

**Comment by melyux** [comment]
Yeah, less than a month. Here are some Moments created by my iPhone automatically:

* Metro Vancouver, February 17, 2023
* Yen-Li Center, January 3, 2023
* Michael's House, New Year's Day

Seems to pretty basic: take clusters of photos taken with a maximum of ~50 miles and ~12 hours gap between photos, and group them together with the date and place. (The most specific place name that applies to ~90% of the photos.)

**Comment by melyux** [comment]
These true Moments would be more of a UI view thing, since they're more utilitarian to create logical division between all photos

**Comment by lastzero** [comment]
I see. We usually travel for more than a day, so this isn't something that seemed to be of great use. The list of day travels in the past 42 years would also be extremely long for me.

**Comment by melyux** [comment]
Quite, I had thought the current Moments could be improved to be more pinpoint, but like you said... now it seems this would be better off as the way to display photos by default (instead of the current unchanging grid). Exactly like this:

![192856972-10e8fecb-f252-4bdd-a8cd-8a5beca0b883](https://user-images.githubusercontent.com/10296053/222826173-845695b5-0cc5-4669-9c28-cf0f4471fef0.png)

**Comment by lastzero** [comment]
OK, I think this is getting all over the place 😂

Also I don't believe this should be our default view. It may make sense for some users and use cases, but I for example have never ever used or enjoyed this once in my entire life.

**Comment by melyux** [comment]
It's just a way of displaying photos. I think most would prefer their photos to be slightly separated out into... moments rather than getting hit by just a grid. Synology, Apple, Google display photos by default like this for this reason. With a grid, two neighboring photos could be years or thousands of miles apart, so it's a lot more difficult to know where/when you are while looking and scrolling

**Comment by lastzero** [comment]
Did I mention that I don't use Google Photos? That's one of the reasons.

**Comment by lastzero** [comment]
Also note we already have an issue for a timeline view. All that's missing is funding for more developers.

**Comment by melyux** [comment]
Haha alright, doesn't have to be the only choice and should be able to be turned off to see the current flat grid. But I'm sure most would prefer some logical separation instead of just a grid

In either case, the current "Moments" could be improved using the same concept at a less granular level (maximum gap of days instead of hours)

**Comment by lastzero** [comment]
In fact, it's the first time someone is asking for this. So I'm sure it's not like all of our 25k+ users are missing it every day

**Comment by melyux** [comment]
It's a quality of life feature that's not advertised, but if you show it side by side to someone along with the flat grid, most would choose the more organized view. It's what all successful photo management software has converged onto

**Comment by lastzero** [comment]
That may be. However, this is what we have in store and we should focus on one thing on this feature request. We are really getting all over the place.

**Comment by graciousgrey** [comment]
> Haha alright, doesn't have to be the only choice and should be able to be turned off to see the current flat grid. But I'm sure most would prefer some logical separation instead of just a grid

The view you are suggesting might be nice as additional view type for the timeline view once it is developed.

I also agree that we can refine the existing moments, so that they also capture weekend trips or holidays.

**Comment by dlgreenwald** [comment]
I shared it in the Show and Tell section of the discussions area, but I have hacked together a script that drives the PhotoPrism API to produce albums like this issue describes.   Right now it's a pair of Python scripts utilizing Pandas to do some bare minimum statistical analysis for anomaly detection but if there is a path towards converting them, or utilizing the same concepts I fleshed out there I would be happy to contribute.

https://github.com/photoprism/photoprism/discussions/3650
